narwhals/_compliant/expr.py

```diff
@@ -64,6 +64,17 @@ class NativeExpr(Protocol):
     def between(self, *args: Any, **kwds: Any) -> Any: ...
     def isin(self, *args: Any, **kwds: Any) -> Any: ...
 
+    # NOTE: None of these are annotated for `dx.Series`, but are added imperatively
+    # Probably better to define a sub-protocol for `NativeSQLExpr`
+    # - match `dx.Series` to `NativeExpr`
+    # - match the others to `NativeSQLExpr`
+    def __gt__(self, value: Any, /) -> Self: ...
+    def __lt__(self, value: Any, /) -> Self: ...
+    def __ge__(self, value: Any, /) -> Self: ...
+    def __le__(self, value: Any, /) -> Self: ...
+    def __eq__(self, value: Any, /) -> Self: ...  # type: ignore[override]
+    def __ne__(self, value: Any, /) -> Self: ...  # type: ignore[override]
```

Comment:
looks good, thanks - if anything, I like it better if we match on these than if we match on `between` and `isin` (which, for example, Daft doesn't have)